tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+**Flexible Builds: `Multi.cproject.yml`**
+
+```yml
+target-types:
+  - type: Board
+    board: NUCLEO-L552ZE-Q
+
+  - type: Production-HW
+    device: STM32L552XY         # specifies device
+
+  - type: Virtual
+    board: VHT-Corstone-300     # Virtual Hardware platform (appears as board)
+      
+build-types:
+  - type: Debug
+    optimize: debug
+    debug: on
+
+  - type: Test
+    optimize: max
+    debug: on
+
+  - type: Release
+    optimize: max
+    debug: off
+
+groups:
+  - group: My group1
+    files:
+      - file: file1a.c
+      - file: file1b.c
+      - file: file1c.c
+
+  - group: My group2
+      - file: file2a.c
+
+  - group: Test-Interface
+    for-type: .Test
+      - file: fileTa.c
+
+layers:
+  - layer: NUCLEO-L552ZE-Q/Board.clayer.yml   # tbd find a better way: i.e. $Board$.clayer.yml
+    for-type: +Board
+
+  - layer: Production.clayer.yml              # added for target type: Production-HW
+    for-type: +Production-HW
+
+  - layer: Corstone-300/Board.clayer.yml      # added for target type: VHT-Corstone-300
+    for-type: +VHT-Corstone-300
+
+components:
+  - component: Device:Startup
+  - component: CMSIS:RTOS2:FreeRTOS
+  - component: ARM::CMSIS:DSP&Source          # not added for build type: Test
+    not-for-type: .Test                           
+```
+
+# Project Setup for Related Projects
+
+A solution is the software view of the complete system. It combines projects that can be generated independently and therefore manages related projects. It may be also deployed to different targets during development as described in the previous section under [Project Setup for Multiple Targets and Test Builds](#project-setup-for-multiple-targets-and-test-builds).
+
+The picture below shows a system that is composed of:
+
+- Project A: that implements a time-critical control algorithm running on a independent processor #2.
+- Project B: which is a diagram of a cloud connected IoT application with Machine Learning (ML) functionality.
+- Project C: that is the data model of the Machine Learning algorithm and separate to allow independent updates.
+- Project D: that implements the device security (for example with TF-M that runs with TrustZone in secure mode).
+
+In addition such systems may have a boot-loader that can be also viewed as another independent project.
+
+![Related Projects of an Embedded System](./images/Solution.png "Related Projects of an Embedded System")
+
+To manage the complexity of such related a projects, the `*.csolution.yml` file is introduced. At this level the `target-types` and `build-types` may be managed, so that a common set is available across the system. However it should be also possible to add project specific `build-types` at project level.  (tdb: `target-types` might be only possible at solution level).
```

Comment:
I don't think the complexity of such project comes from the multiplication of targets and build types. I'd think it rather comes from the projects inter-dependencies: sharing of resources, in particular memory map, or build order, or build pre/post processing 
e.g. what happens if the bootloader size increase, how can solution help the user detect that other projects needs to be relocated 
To address this kind of issue, I would expect an entry in the solution file that is common to several sw projects of the same target-type.  In the example below, if user wants to add MQTT_Azure as an alternative to MQTT_WAS, is there a way to specify that a first "solution" is Bootloader+TFM+MQTT_AWS and an alternative "solution" is Bootloader+TFM+MQTT_Azure  ?